### PR TITLE
Added a notion of 'intent to delegate'.

### PR DESCRIPTION
**Pull request by raymeskhoury**
I attempted to rework delegation section in terms of "intent to delegate". I also collapsed the restrictions
section into the section on delegation because they are intertwined.

Let me know if you think this is an improvement!

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -203,78 +203,76 @@ efficient implementations.</p>
...
-href="#delegating-permissions">Delegating Permissions</a>.
+<h2>Delegating Permissions</h2>
 
-<p>In cases where an embedder has not (yet) delegated a permission, the UA MUST
-prevent the embedee from triggering permission prompts to the user, and the UA
-SHOULD prevent the embedee from acquiring any permissions based on a prior
-decision made by the user.</p>
+<p>An embedder is said to show an <dfn>intent to delegate</dfn> a permission to
+an embedee if it indicates that a permission should be delegated to that
+embedee. An embeder can show <a>intent to delegate</a> regardless of whether or
```
Comment (noncombatant) [inline index.bs]: Typo: embedder

File: index.bs [hunk 2]
```diff
@@ -203,78 +203,76 @@ efficient implementations.</p>
...
+not it actually holds the permission itself. Showing an <a>intent to
+delegate</a> does not mean that the embedee will automatically gain acess to the
```
Comment (noncombatant) [inline index.bs]: Typo: access

File: index.bs [hunk 3]
```diff
@@ -203,78 +203,76 @@ efficient implementations.</p>
...
+require that:</p>
 
-<p>UAs MAY also allow cross-origin embedees to acquire permission grants through
-the use of advanced user configurations, such as custom whitelists (see
-<a href="#reduced-user-control">Reduced User Control</a> for further
-discussion).</p>
+<ol>
 
-</section>
+<li>The embedder holds that permission itself.</li>
```
Comment (noncombatant) [inline index.bs]: Only this? What about, the embedee must qualify for the permission (e.g. secure origin, or whatever else)?

File: index.bs (hunk 1 above)
Comment (raymeskhoury) [inline index.bs]: Done.

File: index.bs (hunk 2 above)
Comment (raymeskhoury) [inline index.bs]: Done.

File: index.bs (hunk 3 above)
Comment (raymeskhoury) [inline index.bs]: Yep, there are 3 points in this list (see below)

**Pull request merged**